pkg/crc/network/utils.go

```diff
@@ -95,23 +95,42 @@ func UriStringForDisplay(uri string) (string, error) {
 	return uri, nil
 }
 
-func CheckCRCLocalDNSReachableFromHost(clusterName, domainName, appsDomainName string) error {
-	ip, err := net.LookupIP(fmt.Sprintf("api.%s.%s", clusterName, domainName))
+func matchIP(ips []net.IP, expectedIP string) bool {
+	for _, ip := range ips {
+		if ip.String() == expectedIP {
+			return true
+		}
+	}
+
+	return false
+}
+func CheckCRCLocalDNSReachableFromHost(clusterName, domainName, appsDomainName, expectedIP string) error {
+	apiHostname := fmt.Sprintf("api.%s.%s", clusterName, domainName)
+	ip, err := net.LookupIP(apiHostname)
 	if err != nil {
 		return err
 	}
-	logging.Debugf("api.%s.%s resolved to %s", clusterName, domainName, ip)
+	logging.Debugf("%s resolved to %s", apiHostname, ip)
+	if !matchIP(ip, expectedIP) {
+		logging.Debugf("%s is expected to resolve to %s", apiHostname, expectedIP)
+		return fmt.Errorf("Invalid IP for %s", apiHostname)
```

Comment:
@cfergeau can we also add what we are expecting and what we got when failure happen otherwise  user have to again run with debug mode to see what going on.